lib/http/form_data/composite_io.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "stringio"
+
+module HTTP
+  module FormData
+    # Provides IO interface across multiple IO objects.
+    class CompositeIO
+      # @param [Array<IO>] ios Array of IO objects
+      def initialize(*ios)
+        @ios = ios.flatten.map { |io| io.is_a?(String) ? StringIO.new(io) : io }
```

Comment:
Also, CompositeIO seems to be be internal class, and I don't see any point in allowing passing *any amount of args... Instead it should simply accept a single argument as flat array, thus here we will just map it.